src/shim/kernel.rs

```diff
@@ -8,74 +8,41 @@ use fvm4::kernel::SyscallError as E4;
 use fvm_shared2::error::ErrorNumber as N2;
 use fvm_shared3::error::ErrorNumber as N3;
 use fvm_shared4::error::ErrorNumber as N4;
-use itertools::Either;
+use num_derive::FromPrimitive;
+use num_traits::FromPrimitive;
 use std::fmt;
+use std::fmt::Debug;
 
 macro_rules! error_number {
     ($($variant:ident),* $(,)?) => {
         #[derive(Debug, Clone)]
-        #[non_exhaustive]
         pub enum ErrorNumber {
             $($variant,)*
-            Unknown(UnknownErrorNumber),
+            Unknown(u32),
```

Comment:
There is a hole in the current design. It's not illegal for me to do this:
```rust
let e = ErrorNumber::Unknown(KnownErrorNumber::ReadOnly as u32);
```
but it would lead to incorrect error messages.

This should at least be documented.
```suggestion
            /// This is to catch `#[non_exhaustive]` upstream errors, and MUST NOT for be constructed
            /// outside of this module.
            Unknown(u32),
```

But you may choose to fix.

Reply:
Agree. Could this kind of construct be forbidden using our linting framework?

Reply:
No, such a lint would require type information, but this kind of thing should be achieved with a struct instead

Reply:
To be clear, I think this is niche enough to just document, but if you wanted to fix it, you'd have to introduce an intermediate :)